### PR TITLE
Update MurmurHash3_x64_128 to use the cuco equivalent implementation

### PR DESCRIPTION
**Pull request by PointKernel**
## Description
This PR modifies MurmurHash3_x64_128 to utilize the cuco equivalent implementation, eliminating duplication.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by karthikeyann — APPROVED** [review]

File: cpp/include/cudf/hashing/detail/murmurhash3_x64_128.cuh [hunk 1]
```diff
@@ -15,211 +15,96 @@
...
+                             key.size_bytes());
 }
 
 template <>
 MurmurHash3_x64_128<numeric::decimal32>::result_type
   __device__ inline MurmurHash3_x64_128<numeric::decimal32>::operator()(
     numeric::decimal32 const& key) const
 {
-  return compute(key.value());
+  return this->compute(key.value());
```
Comment (mythrocks) [inline cpp/include/cudf/hashing/detail/murmurhash3_x64_128.cuh:91]: I've a silly question:  Why does `this` need to be explicitly mentioned here?

Comment (PointKernel) [reply cpp/include/cudf/hashing/detail/murmurhash3_x64_128.cuh:91]: I personally find that using `this` makes the intention to invoke the private member function `compute` clearer.

File: cpp/include/cudf/hashing/detail/murmurhash3_x64_128.cuh [hunk 2]
```diff
@@ -15,211 +15,96 @@
...
-// MurmurHash3 was written by Austin Appleby, and is placed in the public
-// domain. The author hereby disclaims copyright to this source code.
-// Note - The x86 and x64 versions do _not_ produce the same results, as the
-// algorithms are optimized for their respective platforms. You can still
-// compile and run any of them on any platform, but your performance with the
-// non-native version will be less than optimal.
 template <typename Key>
 struct MurmurHash3_x64_128 {
-  using result_type = thrust::pair<uint64_t, uint64_t>;
+  using result_type = cuda::std::array<uint64_t, 2>;
```
Comment (ttnghia) [inline cpp/include/cudf/hashing/detail/murmurhash3_x64_128.cuh:31]: Is there any benefit from this change?

**Review by ttnghia — APPROVED** [review]

**Review by mythrocks — APPROVED** [review]

**Pull request merged**